source/agora/crypto/Hash.d

```diff
@@ -161,6 +171,53 @@ public void hashPart (T) (scope const auto ref T record, scope HashDg hasher)
             hashPart(field, hasher);
 }
 
+/*******************************************************************************
+
+    Encode an unsigned integer to its variable-length binary format
+
+    VarInt Size
+    size <= 0xFC(252)  -- 1 byte   ubyte
+    size <= USHORT_MAX -- 3 bytes  (0xFD + ushort)
+    size <= UINT_MAX   -- 5 bytes  (0xFE + uint)
+    size <= ULONG_MAX  -- 9 bytes  (0xFF + ulong)
+
+    Params:
+        T = Type of unsigned integer to serialize
+        var = Instance of `T` to serialize
+        dg  = Serialization delegate
+
+    Returns:
+        The serialized convert variable length integer
+
+*******************************************************************************/
+
+private void toVarInt (T) (const T var, scope HashDg hasher)
```

Comment:
I thought `toVarInt` already exists? Has it just been moved?

Reply:
This is a copy of the serializer version with the delegate here as HashDg rather than SerializeDg. We could make it common and name the delegate something more general. But then we need to decide where to put it. I suppose in the crypto submodule would be fine. What do you think?

Reply:
We should name it something else then? It's not converting to a var int. Maybe `hashVarInt` is better?

Reply:
Will update.